plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts

```diff
@@ -212,49 +215,46 @@ export class DatabaseEntitiesCatalog implements EntitiesCatalog {
                     }
                   }
 
-                  break;
-                } catch (e) {
-                  if (e instanceof ConflictError && attempt < BATCH_ATTEMPTS) {
-                    this.logger.warn(
-                      `Failed to write batch at attempt ${attempt}/${BATCH_ATTEMPTS}, ${e}`,
+                  if (options?.outputEntities) {
+                    const writtenEntities = await this.database.entities(
```

Comment:
Point of reading the entities back here is to grab all of their relations right? During a dry run we'll only see relations that already exist or are present in the same batch, which is a bit borked.

Perhaps it makes sense to just limit the size of dry runs? Basically limiting them to only be accepted if they can be run in a single batch?

Reply:
Hm. `outputEntities` is per default false. The only "real" user of this is actually the add new location POST handler. And the only consumer of that API is the Backstage frontend, which doesn't render those relations anyway as part of that call.

Should we leave this as is, and make a ticket to solve it in some different way? The effect here is also that entities will have only partial relations for a while as the batches are being written.

We should probably be adding/replacing them by source entity name, not by originating entity, and writing them along with their corresponding "sources" inside this loop. If that's an understandable explanation :)

Reply:
Yep, given that the issue exists before this PR anyway, I think it's fine to merge and think about how we can properly fix it separately.

As discussed, writing by source entity has the potential for deadlock-like situations